src/fastcs_eiger/eiger_controller.py

```diff
@@ -334,34 +329,89 @@ async def queue_update(self, parameters: list[str]):
 
             await self.stale_parameters.set(True)
 
-    @scan(0.1)
     async def update(self):
-        """Periodically check for parameters that need updating from the detector."""
         if not self._parameter_updates:
             if self.stale_parameters.get():
                 await self.stale_parameters.set(False)
-
             return
 
-        # Take a copy of the current parameters and clear. Parameters may be repopulated
-        # during this call and need to be updated again immediately.
         async with self._parameter_update_lock:
             parameters = self._parameter_updates.copy()
             self._parameter_updates.clear()
 
         # Release lock while fetching parameters - this may be slow
         parameter_updates: list[Coroutine] = []
         for parameter in parameters:
-            match getattr(self, parameter):
+            if parameter in IGNORED_KEYS:
+                continue
+            attr_name = _key_to_attribute_name(parameter)
```

Comment:
```suggestion
            attr_name = parameter.attribute_name
```

And then flatten `_key_to_attribute_name` into `attribute_name` as it's not used anywhere else.

Reply:
Annoyingly `parameter` is just a string here as it's handling the response from the detector